climada/hazard/trop_cyclone.py

```diff
@@ -405,7 +405,19 @@ def _apply_criterion(self, criterion, scale):
                 change = chg['change'] * scale
             if select.any():
                 new_val = getattr(haz_cc, chg['variable'])
-                new_val[select] *= change
+                #For sparse matrices *= is ineficient (slow, uses large memory)
+                #Instead use multiply and chunking
```

Comment:
```suggestion
                # 1d-masks like `select` are inefficient for indexing sparse matrices since
                # they are broadcasted densely in the second dimension
```